src/NGen.cpp

```diff
@@ -327,46 +329,109 @@ int main(int argc, char *argv[]) {
+    std::vector<int>& keys = layer_meta_data.get_keys();
+
+    // get the converted time steps for layers
+    double min_time_step;
+    double max_time_step;
+    std::vector<double> time_steps;
+    for(int i = 0; i < keys.size(); ++i)
+    {
+        auto& m_data = layer_meta_data.get_layer(keys[i]);
+        double c_value = UnitsHelper::get_converted_value(m_data.time_step_units,m_data.time_step,"s");
+        time_steps.push_back(c_value);
+
+        if ( i == 0 )
         {
-          response *= (catchment_collection->get_feature(id)->get_property("area_sqkm").as_real_number() * 1000000);
+            min_time_step = max_time_step = c_value;
         }
-        //TODO put this somewhere else as well, for now, an implicit assumption is that a modules get_response returns
-        //m/timestep
-        //since we are operating on a 1 hour (3600s) dt, we need to scale the output appropriately
-        //so no response is m^2/hr...m^2/hr * 1hr/3600s = m^3/hr
-        response /= 3600.0;
-        //update the nexus with this flow
-        for(auto& nexus : features.destination_nexuses(id)) {
-          //TODO in a DENDRIDIC network, only one destination nexus per catchment
-          //If there is more than one, some form of catchment partitioning will be required.
-          //for now, only contribute to the first one in the list
-          nexus->add_upstream_flow(response, id, output_time_index);
-	        break;
+        else
+        {
+            if ( c_value < min_time_step )
+            {
+                min_time_step = c_value;
+            }
+            else if ( c_value > max_time_step)
+            {
+                max_time_step = c_value;
+            }
         }
-      } //done catchments
+    }
+
+    // check that all layers are time steps divide into the max time step evenly
+    // and calculate the time step multiplyer for each layer
+
+    std::unordered_map<int, int> steps_for_level;
+
+    for( int i = 0; i < keys.size(); ++i )
+    {
+      if ( int(max_time_step) % int(time_steps[i]) != 0 )
+      {
+          throw std::runtime_error("Incompatible time steps");
+      }
+      else
+      {
+        int sub_steps = int(max_time_step) / int(time_steps[i]);
+        steps_for_level[keys[i]] = sub_steps;
+      }
+    }
+
+    std::vector<int> output_time_index;
+    output_time_index.resize(keys.size());
+    std::fill(output_time_index.begin(), output_time_index.end(),0);
+
+    //Now loop some time, iterate catchments, do stuff for total number of output times
+    for( int count = 0; count < manager->Simulation_Time_Object->get_total_output_times(); count++) {
+      for ( long lv : features.levels() ) {
+        //std::cout<<"Output Time Index: "<<output_time_index<<std::endl;
+        if(output_time_index[lv]%100 == 0) std::cout<<"Running timestep " << output_time_index[0] <<std::endl;
+          std::string current_timestamp = manager->Simulation_Time_Object->get_timestamp(output_time_index[0]);
```

Comment:
Inadvertent indentation here confused me for a bit... possibly indent 390-391 to match to avoid changing the diff too much for now.

Reply:
We probably want to update 'Simulation_Time_Object' to expect multiple time layers

Reply:
Yes I was just trying to figure out how that fit in... and whether the current assumptions cause any problems.

Reply:
I think what probably makes sense is to have a `Simulation_Time` instance per layer... and possibly rename it (prob later refactor). The instance might even be part of the `Layer_Description` class... and it could know the "global" simulation time and figure out the timestep index information--encapsulating that capability there.